### PR TITLE
reinvent branch: replaced gson with lambda power tools. Added missing CORS headers, and new Search Handler

### PR DESCRIPTION
**Pull request by sercasti**
*Description of changes:*
  a-  replaced gson with lambda power tools. 
  b- Added missing CORS headers to response so user doesn't need to enable cors from API GW.
  c- Added new SearchHandler to showcase the Path parameter feature from Refactor Spaces

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: MonoToMicroLambda/build.gradle [hunk 1]
```diff
@@ -14,12 +14,14 @@ repositories {
 }
 
 dependencies {
-  implementation 'com.amazonaws:aws-lambda-java-core:1.2.1'
-  implementation 'com.amazonaws:aws-lambda-java-events:3.11.0'
-  implementation 'com.google.code.gson:gson:2.9.1'
-
+  implementation 'com.amazonaws:aws-lambda-java-core:1.2.3'
+  implementation 'com.amazonaws:aws-lambda-java-events:3.11.3'
+  implementation 'com.fasterxml.jackson.core:jackson-databind:2.12.6.1'
```
Comment (deki) [inline MonoToMicroLambda/build.gradle]: why not 2.15.3?

File: MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/CreateHandler.java [hunk 2]
```diff
@@ -22,34 +22,37 @@
...
-      User userRequest = gson.fromJson(event.getBody(),User.class);
+      User userRequest = extractDataFrom(event).as(User.class);
       User profile = repository.getOrCreate(userRequest);
       response
-        .withHeaders(Map.of("Access-Control-Allow-Origin","*"))
-        .withBody(gson.toJson(profile))
+        .withHeaders(Map.of("Content-Type", "application/json",
+          "Access-Control-Allow-Headers", "Content-Type, x-requested-with",
+          "Access-Control-Allow-Origin", "*",
+          "Access-Control-Allow-Methods", "DELETE,OPTIONS,POST,GET"))
```
Comment (deki) [inline MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/CreateHandler.java]: why do we need DELETE in here?

File: MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/LoginHandler.java [hunk 3]
```diff
@@ -22,34 +22,37 @@
...
-      User userRequest = gson.fromJson(event.getBody(),User.class);
+      User userRequest = extractDataFrom(event).as(User.class);
       User profile = repository.getByEmail(userRequest.getEmail());
       response
-        .withHeaders(Map.of("Access-Control-Allow-Origin","*"))
-        .withBody(gson.toJson(profile))
+        .withHeaders(Map.of("Content-Type", "application/json",
+          "Access-Control-Allow-Headers", "Content-Type, x-requested-with",
+          "Access-Control-Allow-Origin", "*",
+          "Access-Control-Allow-Methods", "DELETE,OPTIONS,POST,GET"))
```
Comment (deki) [inline MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/LoginHandler.java]: why do we need DELETE in here?

File: MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/SearchHandler.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      return response;
+    }
+
+    User profile = repository.getByEmail(emailPathParamValue);
+    try {
+      response
+        .withHeaders(Map.of("Content-Type", "application/json",
+          "Access-Control-Allow-Headers", "Content-Type, x-requested-with",
+          "Access-Control-Allow-Origin", "*",
+          "Access-Control-Allow-Methods", "DELETE,OPTIONS,POST,GET"))
```
Comment (deki) [inline MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/SearchHandler.java]: why do we need DELETE in here?

File: MonoToMicroLambda/build.gradle (hunk 1 above)
Comment (sercasti) [reply MonoToMicroLambda/build.gradle]: good catch, fixing

File: MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/CreateHandler.java (hunk 2 above)
Comment (sercasti) [reply MonoToMicroLambda/src/main/java/com/monoToMicro/Lambda/CreateHandler.java]: fixed

**Pull request merged**